fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java

```diff
@@ -113,29 +123,70 @@ public Void visitAlterUserStatement(AlterUserStmt stmt, ConnectContext session)
+             * IDENTIFIED WITH
+             */
+            if (!Strings.isNullOrEmpty(stmt.getAuthPlugin())) {
+                stmt.setAuthPlugin(stmt.getAuthPlugin().toUpperCase());
+                if (AuthPlugin.AUTHENTICATION_LDAP_SIMPLE.name().equals(stmt.getAuthPlugin())) {
+                    stmt.setUserForAuthPlugin(stmt.getAuthString());
+                } else if (AuthPlugin.MYSQL_NATIVE_PASSWORD.name().equals(stmt.getAuthPlugin())) {
+                    // in this case, authString is password
+                    stmt.setScramblePassword(analysePassword(stmt.getUserIdent(), stmt.getAuthString(),
+                            stmt.isPasswordPlain()));
+                } else if (AuthPlugin.AUTHENTICATION_KERBEROS.name().equalsIgnoreCase(stmt.getAuthPlugin()) &&
+                        GlobalStateMgr.getCurrentState().getAuth().isSupportKerberosAuth()) {
+                    // In kerberos authentication, userForAuthPlugin represents the user principal realm.
+                    // If user realm is not specified when creating user, the service principal realm will be used as
+                    // the user principal realm by default.
+                    if (stmt.getAuthString() != null) {
+                        stmt.setUserForAuthPlugin(stmt.getAuthString());
                     } else {
-                        scramblePassword = MysqlPassword.checkPassword(stmt.getOriginalPassword());
+                        stmt.setUserForAuthPlugin(Config.authentication_kerberos_service_principal.split("@")[1]);
                     }
-                } catch (DdlException | AnalysisException e) {
-                    // TODO AnalysisException used to raise in all old methods is captured and translated to SemanticException
-                    // that is permitted to throw during analyzing phrase under the new framework for compatibility.
-                    // Remove it after all old methods migrate to the new framework
-                    throw new SemanticException(e.getMessage());
+                } else {
+                    ErrorReport.reportSemanticException(ErrorCode.ERR_AUTH_PLUGIN_NOT_LOADED, stmt.getAuthPlugin());
                 }
-            } else {
-                scramblePassword = new byte[0];
             }
-            stmt.setScramblePassword(scramblePassword);
+            return null;
+        }
+
+        /**
+         * convert password to hashed password
+         */
+        private byte[] analysePassword(UserIdentity userIdent, String originalPassword, boolean isPasswordPlain) {
+            if (Strings.isNullOrEmpty(originalPassword)) {
+                return new byte[0];
+            }
+            try {
+                if (isPasswordPlain) {
+                    // plain password should check for validation & reuse
+                    Auth.validatePassword(originalPassword);
+                    GlobalStateMgr.getCurrentState().getAuth().checkPasswordReuse(userIdent, originalPassword);
```

Comment:
It would be very confusing to check if password reuse when creating a new user, although we happened to allow it right now. If someone changes it in checkPasswordReuse someday, he would never expect that this could cause the failure of creating a new user. I suggest you add a `shouldReuse` flag as an argument.

Reply:
done